tests/std/tests/P0896R4_ranges_alg_find_if/test.cpp

```diff
@@ -2,69 +2,59 @@
 // SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
 #include <algorithm>
-#include <array>
 #include <cassert>
 #include <concepts>
 #include <ranges>
 #include <utility>
 
 #include <range_algorithm_support.hpp>
+using namespace std;
+using P = std::pair<int, int>;
 
-constexpr void smoke_test() {
-    using ranges::find_if, ranges::iterator_t, std::same_as;
-    using P = std::pair<int, int>;
+constexpr auto matches = [](const int val) { return val == 42; };
+constexpr auto equals  = [](auto x) { return [x](auto&& y) { return y == x; }; };
 
-    std::array<P, 3> const pairs = {{{0, 42}, {2, 42}, {4, 42}}};
-    constexpr auto equals        = [](auto x) { return [x](auto&& y) { return y == x; }; };
+// Validate dangling story
+STATIC_ASSERT(same_as<decltype(ranges::find_if(borrowed<false>{}, matches)), ranges::dangling>);
+STATIC_ASSERT(same_as<decltype(ranges::find_if(borrowed<true>{}, matches)), int*>);
 
-    // Validate dangling story
-    STATIC_ASSERT(same_as<decltype(find_if(borrowed<false>{}, equals(42))), ranges::dangling>);
-    STATIC_ASSERT(same_as<decltype(find_if(borrowed<true>{}, equals(42))), int*>);
-
-    for (auto [value, _] : pairs) {
-        {
-            // Validate range overload [found case]
-            auto result = find_if(basic_borrowed_range{pairs}, equals(value), get_first);
-            STATIC_ASSERT(same_as<decltype(result), iterator_t<basic_borrowed_range<P const>>>);
-            assert((*result).first == value);
+struct instantiator {
+    static constexpr P haystack[3] = {{0, 42}, {2, 42}, {4, 42}};
+
+    template <ranges::input_range Read>
+    static constexpr void call() {
+        using ranges::find_if, ranges::iterator_t;
+
+        for (auto [value, _] : haystack) {

```

Comment:
Also pre-existing copying that should be at least `auto&`.

Reply:
```suggestion
        for (const auto& [value, _] : haystack) {
```